### ISSUE TITLE
Why is the normalization of hinge weights conducted twice?

### ISSUE THREAD

**Issue by ZhichaoDuan** [Jize-W/HingeABL#1]
Hi, when I go though the implementation of HingeABL, I found that the normalization of hinge weights is conducted twice instead of once as pointed out in the original paper, is there any specific reason for such detail?
```
p_logits_diff = logits[:, 0].unsqueeze(dim=1) - logits
p_logits_imp = F.relu(p_logits_diff + m)
p_logits_imp = p_logits_imp * labels
p_logits_imp = p_logits_imp[:, 1:]
p_logits_imp = p_logits_imp / (p_logits_imp.sum(dim=1).unsqueeze(dim=1) + 1e-30)
p_logits_imp = p_logits_imp #+ p_mean.unsqueeze(dim=1)
# p_logits_imp = p_logits_imp * labels[:, 1:]
# p_logits_imp = p_logits_imp / (p_logits_imp.sum(dim=1).unsqueeze(dim=1) + 1e-30)

n_logits_diff = logits - logits[:, 0].unsqueeze(dim=1)
n_logits_imp = F.relu(n_logits_diff + m)
n_logits_imp = n_logits_imp * (1 - labels)
n_logits_imp = n_logits_imp[:, 1:]
n_logits_imp = n_logits_imp / (n_logits_imp.sum(dim=1).unsqueeze(dim=1) + 1e-30)
n_logits_imp = n_logits_imp
# n_logits_imp = n_logits_imp * (1 - labels)[:, 1:]
# n_logits_imp = n_logits_imp / (n_logits_imp.sum(dim=1).unsqueeze(dim=1) + 1e-30)
```
The lines commented are the second time the normalization is conducted.

**Comment by Jize-W** [comment]
Hi @ZhichaoDuan ,thank you for your attention to this project! Indeed, the normalization of hinge weights should only be done once. There was a typo in the code, which has now been corrected. Fortunately, the denominator for the second normalization is 1, so this typo does not affect the experimental results.
